akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/SecurityDirectivesSpec.scala

```diff
@@ -12,133 +12,110 @@ import akka.http.scaladsl.server.AuthenticationFailedRejection.{ CredentialsReje
 import akka.testkit.EventFilter
 
 class SecurityDirectivesSpec extends RoutingSpec {
-  val dontBasicAuth = authenticateBasicAsync[String]("MyRealm", _ => Future.successful(None))
-  val dontOAuth2Auth = authenticateOAuth2Async[String]("MyRealm", _ => Future.successful(None))
   val doBasicAuth = authenticateBasicPF("MyRealm", { case Credentials.Provided(identifier) => identifier })
-  val doOAuth2Auth = authenticateOAuth2PF("MyRealm", { case Credentials.Provided(identifier) => identifier })
-  val authWithAnonymous = doBasicAuth.withAnonymousUser("We are Legion")
-
+  val dontBasicAuth = authenticateBasicAsync[String]("MyRealm", _ => Future.successful(None))
```

Comment:
I know this is a Draft, but I'm not sure this does what we are looking for: when an unauthenticated request comes in, this will still reject the request with a challenge, right? I think the desired behavior would be that it passes through in that case.

Reply:
Not sure? There are tests below, which one do you think doesn't do what it should? Do you mean "pass ifUnauthenticated when authentication data is missing"?

Reply:
Ha, indeed that didn't do what I assumed :D